crypto/src/mls/conversation/own_commit.rs

```diff
@@ -11,35 +11,46 @@ use openmls::prelude::{
 };
 
 impl MlsConversation {
-    pub(crate) fn maybe_self_member_commit<'a>(
+    /// Extract the confirmation tag from a public message that is an own commit.
+    ///
+    /// # Arguments
+    ///
+    /// * `own_commit`: The own commit.
+    ///
+    /// returns: Result<&ConfirmationTag, CryptoError>: The confirmation tag from the own commit or an error if the message is not an own commit.
+
+    pub(crate) fn extract_confirmation_tag_from_own_commit<'a>(
         &self,
-        msg: &'a MlsMessageIn,
-    ) -> CryptoResult<Option<&'a ConfirmationTag>> {
-        match msg.body_as_ref() {
+        own_commit: &'a MlsMessageIn,
+    ) -> CryptoResult<&'a ConfirmationTag> {
+        match own_commit.body_as_ref() {
             MlsMessageInBody::PublicMessage(msg) => {
                 let is_commit = matches!(msg.content_type(), ContentType::Commit);
                 let own_index = self.group.own_leaf_index();
                 let is_self_sent = matches!(msg.sender(), Sender::Member(i) if i == &own_index);
-                let is_self_commit = is_commit && is_self_sent;
+                let is_own_commit = is_commit && is_self_sent;
 
-                match is_self_commit.then_some(msg.body()) {
+                match is_own_commit.then_some(msg.body()) {
                     Some(FramedContentBodyIn::Commit(_)) => {
-                        let ct = msg
+                        let confirmation_tag = msg
                             .auth
                             .confirmation_tag
                             .as_ref()
                             .ok_or(CryptoError::InternalMlsError)?;
-                        Ok(Some(ct))
+                        Ok(confirmation_tag)
                     }
-                    _ => Ok(None),
+                    // Not a self commit. Should never be reached if this function
+                    // is called correctly.
+                    _ => Err(CryptoError::ImplementationError),
```

Comment:
Why not use `unreachable!()`? Similarly for the other instance below.

Reply:
We could; however, I thought of a scenario where CC has a bug and this is reached. We we use `unreachable!()`, clients can not recover. With this, they can. And the desciption of the error `"We have done something terribly wrong and it needs to be fixed"` fits well in that case, so I thought it is the right thing to do.

Reply:
That's actually my problem with it. Are we sure that error will eventually get back to us?
If it's something recoverable and the client just proceeds, in most cases we will never get a report, meaning the bug would remain.

Reply:
OK. You convinced me, I'll change it to `unreachable!()`.